components/eamxx/src/share/grid/grids_manager.cpp

```diff
@@ -136,6 +136,18 @@ alias_grid (const std::string& grid_name,
   g->add_alias(grid_alias);
 }
 
+std::set<std::string> GridsManager::
+get_grid_names () const {
+  std::set<std::string> names;
+  if (m_grids.size()==0) {
+    return names;
+  }
+  for (const auto& g : m_grids) {
+    names.emplace(g.second->name());
+  }
+  return names;
+}
```

Comment:
I ended up using this function a lot, but I don't know that I love it. It doesn't take into account aliases. I could instead always loop over the grid manager's grid map and query the names from there.

Reply:
The method is reasonable. Whether to include aliases depends on how you later intend to use this method. But I think the GM's `get_grid` method also scans if the input name is a valid grid alias, so maybe it would make sense to return aliases as well...

Reply:
I use this as a way to look through the FM grids, so I think adding aliases might result in repeated work.